.github/workflows/docker.yml

```diff
@@ -79,12 +112,12 @@ jobs:
             name=altran1502/${{matrix.image}},enable=${{ github.event_name == 'release' }}
           tags: |
             # Tag with branch name
-            type=ref,event=branch
+            type=ref,event=branch,suffix=${{ matrix.device != 'cpu' && '-' || '' }}${{ matrix.device != 'cpu' && matrix.device || '' }}
             # Tag with pr-number
-            type=ref,event=pr
+            type=ref,event=pr,suffix=${{ matrix.device != 'cpu' && '-' || '' }}${{ matrix.device != 'cpu' && matrix.device || '' }}
             # Tag with git tag on release
-            type=ref,event=tag
-            type=raw,value=release,enable=${{ github.event_name == 'release' }}
+            type=ref,event=tag,suffix=${{ matrix.device != 'cpu' && '-' || '' }}${{ matrix.device != 'cpu' && matrix.device || '' }}
+            type=raw,value=release,enable=${{ github.event_name == 'release' }},suffix=${{ matrix.device != 'cpu' && '-' || '' }}${{ matrix.device != 'cpu' && matrix.device || '' }}
```

Comment:
Maybe just add suffix to the matrix and use it here instead of doing all this string manipulation.

Reply:
On a related note, I think it'd be cool if the cpu variants were tagged both with and without the suffix. That way you could set `DEVICE` in `.env` and have it set the right image and service for you.